WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/VariationListViewModel.kt

```diff
@@ -277,31 +279,39 @@ class VariationListViewModel @Inject constructor(
     }
 
     private fun trackWithProductId(event: AnalyticsEvent) {
-        viewState.parentProduct?.let { track(event, mapOf(KEY_PRODUCT_ID to it.remoteId)) }
+        viewState.parentProduct?.let { tracker.track(event, mapOf(KEY_PRODUCT_ID to it.remoteId)) }
     }
 
     fun onAddAllVariationsClicked() {
+        tracker.track(AnalyticsEvent.PRODUCT_VARIATION_GENERATION_REQUESTED)
         val product = viewState.parentProduct ?: return
 
         val variationCandidates = generateVariationCandidates.invoke(product)
 
         if (variationCandidates.size <= GenerateVariationCandidates.VARIATION_CREATION_LIMIT) {
             triggerEvent(ShowGenerateVariationConfirmation(variationCandidates))
         } else {
+            tracker.track(
+                stat = AnalyticsEvent.PRODUCT_VARIATION_GENERATION_LIMIT_REACHED,
+                properties = mapOf(AnalyticsTracker.KEY_VARIATIONS_COUNT to variationCandidates.size)
+            )
             triggerEvent(ShowGenerateVariationsError.LimitExceeded(variationCandidates.size))
         }
     }
 
     fun onGenerateVariationsConfirmed(variationCandidates: List<VariationCandidate>) {
+        tracker.track(AnalyticsEvent.PRODUCT_VARIATION_GENERATION_CONFIRMED)
```

Comment:
❓ One more idea: could you please add here the number of candidates we request variations for? This will help us to better understand the median value of variations that our users generate.
```suggestion
        tracker.track(
            stat = AnalyticsEvent.PRODUCT_VARIATION_GENERATION_CONFIRMED)
            properties = mapOf(AnalyticsTracker.KEY_VARIATIONS_COUNT to variationCandidates.size)
        )
```